src/code/auxiliaries/getPathToPKSimInstallDir.R

```diff
@@ -0,0 +1,138 @@
+#' If more than one matching path is found a warning is produced.
+#'
+#' @examples
+#' path = .getpathToPKSimInstallDir("7.4")
+#' path2 = .getpathToPKSimInstallDir("7.5", "C:/MyOSPFolder/")
+#' 
+.getpathToPKSimInstallDirFromFileSystem <- function(pksim.version, 
+                                                    base.search.folder = Sys.getenv("ProgramW6432")) {
+  pksim.version <- trimws(pksim.version)
+  base.search.folder <- trimws(base.search.folder)
+  base.search.folder <- normalizePath(base.search.folder)
+  
+  if (.Platform$OS.type != "windows")
+    stop("Only Windows platforms are supported")
+  
+  if (!nzchar(base.search.folder))
+    return(NA)
+    
+  # First guess: OSP/PK-Sim folder
+  suite.name <- "Open Systems Pharmacology"
+  product.name <- "PK-Sim"
+
+  full.guess <- file.path(base.search.folder, suite.name, fsep="\\") 
+  full.match <- dir(full.guess, pattern = product.name, full.names = TRUE, 
+                    include.dirs = TRUE)
+  if (length(full.match) == 0) {
+    # Second guess: Search base folder recursivly for exe.name
+    # This might be expensive !
+    exe.name = "PKSim.exe$"
+    full.match <- list.files(base.search.folder, pattern = exe.name, 
+                             recursive = TRUE, full.names = TRUE)
+    full.match <- dirname(full.match)
+  }
+  
+  if (length(full.match) != 0) {
+    # match version
+    full.match <- grep(pksim.version, full.match, fixed=TRUE, value=TRUE)
+    if (length(full.match) > 1)
+      warning("Ambiguous matches for PK-Sim installation path found.")
+    
+    return(.homogenizePath(full.match[1]))
+  }
+  
+  return(NA)
+}
+
+#' Tries to find the installation path for a specific version of PK-Sim.
+#' Searching is done in the following order:
+#' 1. Search via Windows registry entry 
+#' 2. Search via filesystem in a guessed installation folder from the base.search.folder (fallback 1)
+#' 3. Search via filesystem for PKSim.exe recursivly from the defined base.search.folder (fallback 2)
+#'
+#' @param pksim.version The version number of Pk-Sim as a string.
+#' @param base.search.folder The base folder for filesystem-lookup fallback (default: 64-bit program folder)
+#'
+#' @return
+#' The path to the PK-Sim installation for version pksim.version or NA if no path could be found.
+#' The path is separated with slashes (unix-style) and in compilance with base-R without a trailing slash.
+#' If more than one matching path is found a warning is produced.
+#'
+#' @examples
+#' path = .getpathToPKSimInstallDir("7.4")
+#' path2 = .getpathToPKSimInstallDir("7.5", "C:/MyOSPFolder/")
+#' 
+.getpathToPKSimInstallDir <- function(pksim.version, 
+                                      base.search.folder = Sys.getenv("ProgramW6432")) {
+  
+  reg.path <- .getpathToPKSimInstallDirFromRegistry(pksim.version)
+  if (!is.na(reg.path))
+    return(reg.path)
+  
+  reg.path <- .getpathToPKSimInstallDirFromFileSystem(pksim.version, base.search.folder)
```

Comment:
Please rename `reg.path` to something more general (`pksim.path`?), as it sounds like "Path retrieved from registry", but in the second (and potentially third) case it is retrieved by other methods (file system or NA).

Reply:
Makes sense and is part of current commit.